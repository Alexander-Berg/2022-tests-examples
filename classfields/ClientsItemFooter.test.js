const React = require('react');
const ClientsItemFooter = require('./ClientsItemFooter');
const { shallow } = require('enzyme');
const { shallowToJson } = require('enzyme-to-json');

const createContextProvider = require('autoru-frontend/mocks/createContextProvider').default;
const ContextProvider = createContextProvider({
    linkCabinet: () => 'linkToCabinet',
    linkOld: () => 'linkOld',
});

it('Должен вернуть набор кнопок для объявления', () => {
    const client = {
        id: 'clientId',
    };

    const clientsItemFooter = shallowToJson(shallow(
        <ContextProvider>
            <ClientsItemFooter
                client={ client }
                userId="userId"
                offersCount={ 45 }
                isFreezed={ false }
            />
        </ContextProvider>,
    ).dive());

    expect(clientsItemFooter).toMatchSnapshot();
});

it('deactivateClient: должен вызвать changeStatus с набором параметров', () => {
    const changeStatus = jest.fn();
    const client = {
        agent_id: 'agentId',
        id: 'clientId',
    };

    const clientsItemFooterInstance = shallow(
        <ContextProvider>
            <ClientsItemFooter
                agentId="agentId"
                client={ client }
                userId="userId"
                offersCount={ 45 }
                isFreezed={ false }
                changeStatus={ changeStatus }
            />
        </ContextProvider>,
    ).dive().instance();

    clientsItemFooterInstance.deactivateClient();

    expect(changeStatus).toHaveBeenCalledWith({
        userId: 'userId',
        agency_id: 'agentId',
        client_id: 'clientId',
        status: 'stopped',
    });
});

it('activateClient: должен вызвать changeStatus с набором параметров', () => {
    const changeStatus = jest.fn();
    const client = {
        agent_id: 'agentId',
        id: 'clientId',
    };

    const clientsItemFooterInstance = shallow(
        <ContextProvider>
            <ClientsItemFooter
                agentId="agentId"
                client={ client }
                userId="userId"
                offersCount={ 45 }
                isFreezed={ false }
                changeStatus={ changeStatus }
            />
        </ContextProvider>,
    ).dive().instance();

    clientsItemFooterInstance.activateClient();

    expect(changeStatus).toHaveBeenCalledWith({
        userId: 'userId',
        agency_id: 'agentId',
        client_id: 'clientId',
        status: 'active',
    });
});

describe('renderActionButton тесты', () => {
    it('должен вернуть кнопку декативации, если client.status === active', () => {
        const client = {
            status: 'active',
        };

        const clientsItemFooterInstance = shallow(
            <ContextProvider>
                <ClientsItemFooter
                    client={ client }
                />
            </ContextProvider>,
        ).dive().instance();

        expect(clientsItemFooterInstance.renderActionButton()).toMatchSnapshot();
    });

    it('должен вернуть кнопку активации, если client.status === stopped', () => {
        const client = {
            status: 'stopped',
        };

        const clientsItemFooterInstance = shallow(
            <ContextProvider>
                <ClientsItemFooter
                    client={ client }
                />
            </ContextProvider>,
        ).dive().instance();

        expect(clientsItemFooterInstance.renderActionButton()).toMatchSnapshot();
    });

    it('должен вернуть пустой контейнер для других статусов клиентов', () => {
        const client = {
            status: 'disabled',
        };

        const clientsItemFooterInstance = shallow(
            <ContextProvider>
                <ClientsItemFooter
                    client={ client }
                />
            </ContextProvider>,
        ).dive().instance();

        expect(clientsItemFooterInstance.renderActionButton()).toMatchSnapshot();
    });
});

it('renderMimicButton: должен нарисовать Button, если у клиента несколько юзеров', () => {
    const client = {
        users: [
            { id: 111, email: '111@yandex.ru' },
            { id: 222, email: '222@yandex.ru' },
        ],
    };

    const clientsItemFooterInstance = shallow(
        <ContextProvider>
            <ClientsItemFooter
                client={ client }
            />
        </ContextProvider>,
    ).dive().instance();

    expect(clientsItemFooterInstance.renderMimicButton()).toMatchSnapshot();
});
