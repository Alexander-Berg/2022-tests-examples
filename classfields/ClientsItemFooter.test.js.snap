// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderActionButton тесты должен вернуть кнопку активации, если client.status === stopped 1`] = `
<Button
  className="ClientsItemFooter__button ClientsItemFooter__button_action"
  color="white"
  onClick={[Function]}
  size="l"
  type="button"
  width="default"
>
  Активировать
</Button>
`;

exports[`renderActionButton тесты должен вернуть кнопку декативации, если client.status === active 1`] = `
<Button
  className="ClientsItemFooter__button ClientsItemFooter__button_action"
  color="white"
  onClick={[Function]}
  size="l"
  type="button"
  width="default"
>
  Остановить размещение
</Button>
`;

exports[`renderActionButton тесты должен вернуть пустой контейнер для других статусов клиентов 1`] = `
<div
  className="ClientsItemFooter__button ClientsItemFooter__button_action"
/>
`;

exports[`renderMimicButton: должен нарисовать Button, если у клиента несколько юзеров 1`] = `
<Dropdown
  autoClose={true}
  closeOnInside={true}
  directions={
    Array [
      "left-center",
    ]
  }
  disabled={false}
  switcher={
    <Button
      className="ClientsItemFooter__button ClientsItemFooter__button_mimic"
      color="white"
      size="l"
      type="button"
      width="default"
    >
      <IconSvg
        id="svgID"
        size="24"
      />
    </Button>
  }
  type="click"
>
  <Menu
    maxHeight={null}
    onChange={[Function]}
    onFocusChange={[Function]}
    onItemClick={[Function]}
    size="m"
    tabIndex={0}
  >
    <Item
      className="ClientsItemFooter__item"
      id={111}
      value={111}
    >
      <Button
        clasName="ClientsItemFooter__button"
        size="l"
        type="link"
        url="linkOld"
        width="default"
      >
        111@yandex.ru
      </Button>
    </Item>
    <Item
      className="ClientsItemFooter__item"
      id={222}
      value={222}
    >
      <Button
        clasName="ClientsItemFooter__button"
        size="l"
        type="link"
        url="linkOld"
        width="default"
      >
        222@yandex.ru
      </Button>
    </Item>
  </Menu>
</Dropdown>
`;

exports[`Должен вернуть набор кнопок для объявления 1`] = `
<div
  className="ClientsItemFooter"
>
  <Button
    className="ClientsItemFooter__button"
    color="white"
    disabled={false}
    size="l"
    type="link"
    url="linkToCabinet"
    width="full"
  >
    Объявления 
    <span>
      45
    </span>
  </Button>
  <Button
    className="ClientsItemFooter__button"
    color="white"
    disabled={false}
    size="l"
    type="link"
    url="linkToCabinet"
    width="full"
  >
    Кошелек
  </Button>
  <Button
    className="ClientsItemFooter__button"
    color="white"
    disabled={false}
    size="l"
    type="link"
    url="linkToCabinet"
    width="full"
  >
    Настройка тарифа
  </Button>
  <Button
    className="ClientsItemFooter__button"
    color="white"
    disabled={false}
    size="l"
    type="link"
    url="linkToCabinet"
    width="full"
  >
    Информация о салоне
  </Button>
  <Button
    className="ClientsItemFooter__button"
    color="white"
    disabled={false}
    size="l"
    type="link"
    url="linkToCabinet"
    width="full"
  >
    Реквизиты
  </Button>
  <div
    className="ClientsItemFooter__button ClientsItemFooter__button_action"
  />
  <Button
    className="ClientsItemFooter__button ClientsItemFooter__button_mimic"
    color="white"
    size="l"
    type="link"
    url=""
    width="default"
  >
    <IconSvg
      id="svgID"
      size="24"
    />
  </Button>
</div>
`;
